rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py

```diff
@@ -0,0 +1,186 @@
+        distribution_system.get("drain_heat_recovery_efficiency")
+        if (swh_use.get("is_heat_recovered_by_drain"))
+        else 0
+    )
+    use_units = swh_use.get("use_units")
+    assert_(use_units is not None, "A use unit must be provided")
+
+    space_ids = get_spaces_served_by_swh_use(rmd, swh_use["id"])
+    space_within_building_segment_ids = [
+        space["id"] for space in find_all("$.zones[*].spaces[*]", building_segment)
+    ]
+    spaces = [
+        find_exactly_one_with_field_value(
+            "$.buildings[*].building_segments[*].zones[*].spaces[*]",
+            "id",
+            space_id,
+            rmd,
+        )
+        for space_id in space_ids
+        if space_id in space_within_building_segment_ids
+    ]
+
+    if len(spaces) == 0 and use_units not in REQUIRED_USE_UNIT:
+        spaces = find_all("$.zones[*].spaces[*]", building_segment)
+
+    energy_required_by_space = {}
+    volume_flow_rate = 0
+    hourly_values = (
+        hourly_schedule.get("hourly_values")
+        if hourly_schedule is not None
+        else [1] * len(inlet_temperature_schedule)
+    )
+    for space in spaces:
+        if use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_PERSON:
+            power = swh_use_value * space.get("number_of_occupants", 0)
+            energy_required = (
+                power
+                * sum(hourly_values)
+                * (1 - drain_heat_recovery_efficiency)
+                * ureg("Btu")
+            )
+            energy_required_by_space[space["id"]] = energy_required
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_AREA:
+            power = swh_use_value * space.get("floor_area", ZERO.AREA).magnitude
+            energy_required = (
+                power
+                * sum(hourly_values)
+                * (1 - drain_heat_recovery_efficiency)
+                * ureg("Btu")
+            )
+            energy_required_by_space[space["id"]] = energy_required
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER:
+            energy_required_by_space[space["id"]] = (
+                swh_use_value
+                * sum(hourly_values)
+                * (1 - drain_heat_recovery_efficiency)
+                * ureg("Btu")
+            )
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.VOLUME_PER_PERSON:
+            volume_flow_rate += swh_use_value * space.get("number_of_occupants", 0)
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.VOLUME_PER_AREA:
+            volume_flow_rate += (
+                swh_use_value * space.get("floor_area", ZERO.AREA).magnitude
+            )
+
+        elif use_units == SERVICE_WATER_HEATING_USE_UNIT.VOLUME:
+            volume_flow_rate += swh_use_value
+
+        else:
+            energy_required_by_space[space["id"]] = "UNDETERMINED"
+
+        if energy_required_by_space.get(space["id"]) is None:
+            energy_required = 0 * ureg("Btu")
```

Comment:
To be consistent, we should use ```ZERO.ENERGY```. If this doesn't exist, you can create it in the ```ZERO``` class.

Reply:
There is no ZERO.ENERGY in the unit system. https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/utils/pint_utils.py

Reply:
You can add a new one in line 96. 
```Python
ENERGY: Quantity = 0 * ureg("Btu")
```